### PR TITLE
feat: center Gantt view on today by default

### LINKED ISSUE #1671: feat: Center Gantt chart view on current day by default
### Problem Statement

The Gantt chart is a very useful feature. However, when opening the Gantt view, the layout does not automatically center on the current day. For long-running projects with many past tasks, it can be tedious to manually scroll horizontally every time to locate "today".

### Proposed Solution

When navigating to the Gantt chart view, the grid should automatically scroll horizontally to bring the current day column into view.

A potential implementation could be targeting the element representing "today" (or the first active task for the current day) and calling `.scrollIntoView` on component mount. However, I have little to no UI experience, so I don't know if this is the right approach. 

### Alternative Solutions

- Jump to Today button in the header toolbar
- Hide completed/past tasks to reduce horizontal (and vertical) scroll distance
- Leave it up to the user to regularly archive completed/past tasks. 

### Relevant Context

Tested on Kaneo v2.22 (Docker container) on Firefox and Chromium.

### Does this feature align with Kaneo's focus on simplicity?

Yes, it improves use experience without adding new features or controls.

### PR DESCRIPTION
**Pull request by reachsanjivbhagat-gif**
## Description
As a PM, the Gantt view is usually the first screen I open in the morning to see where things stand. Right now it always renders starting from the earliest task in the project, so on any project that's been running a while I have to manually scroll forward to find today before I can actually see what's due this week. This PR makes the timeline center on today by default when the view loads, and adds a small "Jump to Today" button in the toolbar so you can get back to it after scrolling around to look at earlier or later work. The button is disabled when nothing is scheduled near today, since scrolling to nothing wouldn't help anyone.

## Related Issue(s)
Fixes #1671

## Type of Change
- [x] New feature (non-breaking change that adds functionality)

## How Has This Been Tested?
- [x] Unit tests
- [x] Manual testing

Added `gantt.test.tsx` covering both cases: the button staying disabled with no auto-scroll when nothing is scheduled near today, and the auto-center-on-load plus manual re-center behavior when a task spans today. Ran `tsc --noEmit`, `vitest run`, and `biome check` for the affected package and they're all clean. I didn't do a full production build of the whole monorepo (a couple of unrelated internal workspace packages aren't available to me locally), but the web app's own type-check, test suite, and linter all pass.

## Screenshots (if applicable)
N/A - this is a behavioral change, not a visual one. The Gantt view looks the same, it just opens scrolled to today's date instead of the start of the project.

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I understand and take responsibility for every change, and I wrote this pull request description in my own words
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

## Additional Notes
Also added the `jumpToToday` translation key across every locale file so the button label stays consistent wherever the project is used - real translations where the file was already fully translated, English left as a placeholder for the couple of locales still catching up (which the contributing guide says is fine to do).

## Summary by CodeRabbit

- **New Features**
  - Added a “Jump to today” control to the Gantt view.
  - The timeline automatically centers on today when appropriate, including after switching projects.
  - Manual scrolling accounts for the sticky task rail.
  - The control is disabled when no filtered tasks are available.
  - Added labels for the control across supported languages.

- **Tests**
  - Added coverage for automatic centering, project switching, scroll alignment, and disabled states.

### CODE REVIEW

File: apps/web/src/routes/_layout/_authenticated/dashboard/workspace/$workspaceId/project/$projectId/gantt.tsx [hunk 1]
```diff
@@ -168,6 +198,15 @@ function RouteComponent() {
     return () => observer.disconnect();
   }, [timeline]);
 
+  // Center the view on today the first time it becomes available, so opening
+  // the Gantt chart on a long-running project doesn't drop you at the start
+  // of the timeline with today scrolled off-screen.
+  useLayoutEffect(() => {
+    if (hasCenteredOnTodayRef.current || !todayInRange) return;
+    hasCenteredOnTodayRef.current = true;
+    scrollToToday("auto");
```
[the remark this answers is not in the document]
Comment (reachsanjivbhagat-gif) [reply apps/web/src/routes/_layout/_authenticated/dashboard/workspace/$workspaceId/project/$projectId/gantt.tsx]: Leaving this one as-is rather than extending the range. The timeline's bounds are intentionally task-driven (see the "Week-aligned bounds around task dates" comment above `timeline`), and `todayInRange` is meant to reflect whether today is genuinely relevant to what's currently scheduled. A project made up entirely of past or far-future tasks has no "today" column to speak of - that's exactly the case the disabled "Jump to today" button and the `!todayInRange` branch already handle deliberately, and there's an existing passing test ("disables the button and never auto-scrolls when no task falls near today") that codifies this as the intended behavior, not a gap.

Padding the range just to force today into view for an inactive/completed project would mean showing a mostly-empty timeline stretching out to the current date for no reason tied to the actual work. If there's a real product need for "always show today regardless of task dates," I think that's a separate, deliberate design decision (e.g. always including today in the week-aligned bounds) rather than a fix to this PR's guard - happy to open a follow-up issue for it if that's wanted.

File: apps/web/src/routes/_layout/_authenticated/dashboard/workspace/$workspaceId/project/$projectId/gantt.tsx [hunk 2]
```diff
@@ -197,6 +270,17 @@ function RouteComponent() {
               />
             </div>
 
+            <Button
+              variant="outline"
+              size="xs"
+              className="min-h-11 touch-manipulation sm:min-h-0"
+              onClick={() => scrollToToday()}
+              disabled={!todayInRange || scheduledTasks.length === 0}
```
[the remark this answers is not in the document]
Comment (reachsanjivbhagat-gif) [reply apps/web/src/routes/_layout/_authenticated/dashboard/workspace/$workspaceId/project/$projectId/gantt.tsx:284]: Good catch on the distinction, but I'm going to leave this at the narrower fix for now. `scheduledTasks.length === 0` covers the concrete regression the new test exercises: a search that filters out every task while today still falls in the unfiltered range. The deeper case you're describing - some filtered tasks remain, but none of them are near today, while a hidden task keeps the range wide - is real, but it needs a definition of "near today" that doesn't exist anywhere else in this component (the timeline only tracks a coarse date range, not a "near" window), so building it now would mean inventing a new threshold rather than fixing an existing one. That feels like its own follow-up rather than something to bolt onto this predicate speculatively. Happy to open an issue for it if you think it's worth tracking.

File: apps/web/src/routes/_layout/_authenticated/dashboard/workspace/$workspaceId/project/$projectId/gantt.test.tsx [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+    // has nothing left to scroll to and must reflect that.
+    fireEvent.change(
+      screen.getByPlaceholderText("Search scheduled tickets..."),
+      {
+        target: { value: "no such task" },
+      },
+    );
+
+    expect(
+      screen.getByText('No scheduled tasks match "no such task"'),
```
[the remark this answers is not in the document]
Comment (reachsanjivbhagat-gif) [reply apps/web/src/routes/_layout/_authenticated/dashboard/workspace/$workspaceId/project/$projectId/gantt.test.tsx:294]: Leaving this as single-quoted. The string itself contains a double-quoted phrase ("no such task"), so switching the outer literal to double quotes means escaping those inner quotes, per the proposed fix. This repo's biome.json sets `javascript.formatter.quoteStyle: "double"`, but Biome's formatter (like Prettier) picks whichever quote avoids extra escaping when a string contains the other quote character - it doesn't force double quotes unconditionally. `biome check` already passes clean on this line as-is, and running the formatter on the escaped version would just flip it back to single quotes. Resolving as a non-issue given that.

**Comment by tinsever** [comment]
Fixed the filtered-project navigation edge case in 010c72c9a. The new regression test failed before the fix and now passes; all 160 web tests and web typecheck pass. Auto-centering is only consumed after the visible today cell exists.
